Rules/UseConsistentWhitespace.cs

```diff
@@ -211,6 +235,112 @@ private IEnumerable<DiagnosticRecord> FindOpenBraceViolations(TokenOperations to
+                        GetError(ErrorKind.AfterOpeningBrace),
+                        lCurly.Value.Extent,
+                        GetName(),
+                        GetDiagnosticSeverity(),
+                        tokenOperations.Ast.Extent.File,
+                        null,
+                        GetCorrections(lCurly.Previous.Value, lCurly.Value, lCurly.Next.Value, true, false).ToList());
+                }
+            }
+
+            foreach (var rCurly in tokenOperations.GetTokenNodes(TokenKind.RCurly))
+            {
+                if (rCurly.Previous == null
+                    || !IsPreviousTokenOnSameLine(rCurly)
+                    || rCurly.Previous.Value.Kind == TokenKind.LCurly
+                    || rCurly.Previous.Value.Kind == TokenKind.NewLine
+                    || rCurly.Previous.Value.Kind == TokenKind.LineContinuation
+                    || ((rCurly.Previous.Value.TokenFlags & TokenFlags.MemberName) == TokenFlags.MemberName))
+                {
+                    continue;
+                }
+
+                if (!IsPreviousTokenApartByWhitespace(rCurly))
+                {
+                    yield return new DiagnosticRecord(
+                        GetError(ErrorKind.BeforeClosingBrace),
+                        rCurly.Value.Extent,
+                        GetName(),
+                        GetDiagnosticSeverity(),
+                        tokenOperations.Ast.Extent.File,
+                        null,
+                        GetCorrections(rCurly.Previous.Value, rCurly.Value, rCurly.Next.Value, false, true).ToList());
+                }
+            }
+        }
+
+        private IEnumerable<DiagnosticRecord> FindPipeViolations(TokenOperations tokenOperations)
+        {
+            foreach (var pipe in tokenOperations.GetTokenNodes(TokenKind.Pipe))
+            {
+                if (pipe.Next == null
+                    || !IsPreviousTokenOnSameLine(pipe)
+                    || pipe.Next.Value.Kind == TokenKind.Pipe
+                    || pipe.Next.Value.Kind == TokenKind.NewLine
+                    || pipe.Next.Value.Kind == TokenKind.LineContinuation
+                    || ((pipe.Next.Value.TokenFlags & TokenFlags.MemberName) == TokenFlags.MemberName))
+                {
+                    continue;
+                }
+
+                if (!IsNextTokenApartByWhitespace(pipe))
+                {
+                    yield return new DiagnosticRecord(
+                        GetError(ErrorKind.AfterPipe),
+                        pipe.Value.Extent,
+                        GetName(),
+                        GetDiagnosticSeverity(),
+                        tokenOperations.Ast.Extent.File,
+                        null,
+                        GetCorrections(pipe.Previous.Value, pipe.Value, pipe.Next.Value, true, false).ToList());
+                }
+            }
+
+            foreach (var pipe in tokenOperations.GetTokenNodes(TokenKind.Pipe))
```

Comment:
can these loops be combined in some way?

Reply:
Tricky, are you afraid in terms of performance because we call `tokenOperations.GetTokenNodes` foreach rule setting separately? I am not sure if querying for all token kinds once and then putting them into a separate list would really be faster but I am not an expert at performance optimisation.